### PR TITLE
New CSOM doc for retention labels

### PR DESCRIPTION
**Pull request by kyracatwork**
## Category

- [ ] Content fix
- [x ] New article

## What's in this Pull Request?
New doc advising recommend use of CSOM methods for retention label features.
From feature PM 

## Please describe the changes in this PR. Sample description or details around bugs which are being fixed.*
Information on existing methods available in prod, summarized
N/A

### CODE REVIEW

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Retention labels may classify contents as records, which place restrictions on what actions are allowed or blocked. [Learn more about declaring records by using retention labels](/microsoft-365/compliance/declare-records)
+
+CSOM methods are available to apply (set) a retention label (ComplianceTag) on one or many items (ListItems) in SharePoint. 
+
+> [!IMPORTANT]
+> Some information relates to prerelease product that may be substantially modified before it’s released. Microsoft makes no warranties, express or implied, with respect to the information provided here.
+
+## SetComplianceTagOnBulkItems
+This method can be used to set a ComplianceTag on one or many ListItems. It is strongly recommended to use this method for this purpose.
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:21]: Headings must have line breaks around them.

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+<!-- {
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:23]: Not sure what this is, but it's not valid for docs. Please remove this HTML comment from the doc.

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+-->
+```c#
+public List<int> SetComplianceTagOnBulkItems( 
+             List<int> itemIds, 
+             string listUrl, 
+             string complianceTagValue)
+```
+
+### Parameters
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:34]: Line breaks around headings

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+'List' [Int](/en-us/dotnet/api/system.int32)
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:35]: If this is a list, please convert to bullet list.

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+### Applies to
+
+|Product|Versions|
+|:---|:---|
+|SharePoint CSOM|latest|
+
+
+
+## Other
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:53]: line break around headings

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+> [!NOTE] It is strongly recommended to use SetComplianceTagOnBulkItems instead of these methods.
+
+The following methods are also available, but are no longer updated and may be subject to deprecation in the future. If you are using these methods, we strongly recommend use of the SetComplianceTagOnBulkItems method instead.
+
+* [SetComplianceTag](/dotnet/api/microsoft.sharepoint.client.listitem.setcompliancetag?view=sharepoint-csom)
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:58]: bullet points shouldn't have line breaks around them

**Review by andrewconnell — CHANGES_REQUESTED** [review]
The file referenced in the TOC submission doesn't match the file in the PR. The file in the PR is the problem... it should be a markdown file with no spaces. Please rename the content page & address other change requests before we can process this PR.

File: docs/sp-add-ins/CSOM methods for applying retention labels (hunk 4 above)
Comment (kyracatwork) [reply docs/sp-add-ins/CSOM methods for applying retention labels:35]: Actually I just realized there was another interpretation. I'll reach out directly.

File: docs/sp-add-ins/CSOM methods for applying retention labels [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+### Applies to
+
+|Product|Versions|
+|:---|:---|
+|SharePoint CSOM|latest|
+
+## Other
+
+> [!NOTE] It is strongly recommended to use SetComplianceTagOnBulkItems instead of these methods.
```
Comment (andrewconnell) [inline docs/sp-add-ins/CSOM methods for applying retention labels:52]: Invalid note format... please see how it's done elsewhere in this doc. The `> [!NOTE]` must be on it's own line

**Review by andrewconnell — CHANGES_REQUESTED** [review]
Still not a valid file. Must be a markdown file.

**Review by andrewconnell — CHANGES_REQUESTED** [review]
I've fixed the content issues in the doc & addressed all warnings. But I can't move the file. Once the file is moved, it will fix the last automated build error report.

WHAT'S WRONG: The file was saved to the `/sp-add-ins` folder, but the TOC link points to a different location, meaning this will result in a 404 in the nav.

WHAT NEEDS TO BE DONE: In your PR, please move the file > `/docs/apis` folder. That's where general API content should go. This is about the CSOM & not related to SP Add-ins so it's currently in the wrong spot.

**Comment by kyracatwork** [comment]
@andrewconnell moved the file! Thanks for the super-clear instructions.

**Review by andrewconnell — CHANGES_REQUESTED** [review]
I renamed the file in my last commit, but you reverted that. The file name you've created is _not_ valid (spaces in it)... I made the whole lcase & added hyphens. Please fix this.

Also, remove the change to the **.gitignore**... that is not relevant.

**Comment by kyracatwork** [comment]
@andrewconnell I'm still not sure how I did that to the gitignore in the first place. Yikes. Thanks for your help. 

For the file re-name, do I need to change this one line 2 as well or is the filename sufficient?

**Comment by andrewconnell** [comment]
@kyracatwork the only thing that appears broken right now is the fact the *.md file contains `ms.prod` in the frontmatter. That's not needed anymore now that it's in the `api` folder because it's being set by a global config to all files in that folder. So by having it in the file, it's getting double up'd. 

If you can remove that `ms.prod` line from the file & push the commit, it should pass all build checks.

**Review by andrewconnell — APPROVED** [review]

**Pull request merged**